internal/c/start.go

```diff
@@ -167,11 +170,9 @@ func (chSpec ChildSpec) DoStart(
 		// block and wait here until an error (or lack of) is reported from the
 		// client code
 		err := chSpec.Start(childCtx, func(err error) {
-			// we tell the spawner this child thread has started running
```

Comment:
What motivated this change? 

The previous and current implementation may have a similar outcome, but it truly depends on how the reader deals with messages. If there is a `_, ok := <-startChan` on the reader side, this may not behave the same way.

Reply:
Ah, I see, we have panics on the second `startChan<-` call in the defer we are introducing, this makes sense.  My concerns are not that important here, as this context controls the read/write of the `startCh`